include/istio/control/http/check_data.h

```diff
@@ -38,8 +38,11 @@ class CheckData {
   // Get downstream tcp connection ip and port.
   virtual bool GetSourceIpPort(std::string *ip, int *port) const = 0;
 
-  // If SSL is used, get origin user name.
-  virtual bool GetSourceUser(std::string *user) const = 0;
+  // If SSL is used, get peer SAN URI.
+  virtual bool GetPeerPrincipal(std::string *user) const = 0;
```

Comment:
How about
GetPricipal(bool peer, ....)

Reply:
Hmm, local is not necessarily an opposite of peer. I think it might be possible to have only one set but not the other. How do we distinguish this case?

Reply:
What I means is:  

GetPrincipal(bool peer, string* user);

You still call it twice:

GetPrincipal(true, user);  // for peer
GetPrncipal(false, user); // for local.

The whole idea is to re-use code.  but not good for code readability.

Reply:
If use enum type, readability improves.

Reply:
done, PTAL